moveit_fake_controller_manager/src/moveit_fake_controller_manager.cpp

```diff
@@ -236,14 +92,22 @@ class MoveItFakeControllerManager : public moveit_controller_manager::MoveItCont
 
         if (controller_list[i]["joints"].getType() != XmlRpc::XmlRpcValue::TypeArray)
         {
-          ROS_ERROR_STREAM("MoveItFakeControllerManager: The list of joints for controller " << name << " is not specified as an array");
+          ROS_ERROR_STREAM("MoveItFakeControllerManager: The list of joints for controller "
+                           << name << " is not specified as an array");
           continue;
         }
         std::vector<std::string> joints;
-        for (int j = 0 ; j < controller_list[i]["joints"].size() ; ++j)
+        for (int j = 0; j < controller_list[i]["joints"].size(); ++j)
           joints.push_back(std::string(controller_list[i]["joints"][j]));
 
-        controllers_[name].reset(new FakeControllerHandle(name, node_handle_, joints));
+        const std::string &type =
+            controller_list[i].hasMember("type") ? std::string(controller_list[i]["type"]) : DEFAULT_TYPE;
+        if (type == "last point")
+          controllers_[name].reset(new LastPointController(name, joints, pub_));
+        else if (type == "via points")
+          controllers_[name].reset(new ViaPointController(name, joints, pub_));
+        else
+          controllers_[name].reset(new InterpolatingController(name, joints, pub_));
```

Comment:
i prefer to check for exact string ("interpolate") and have the else statement throw an error in case user did something odd like misspell "last point" - they wouldn't expect to get "interpolate"